File: 2389.longestSubsequenceWithLimitedSum.java
import java.util.Arrays;

class Solution {
    public int[] answerQueries(int[] nums, int[] queries) {
        Arrays.sort(nums);
        int results[] = new int[queries.length];
        for (int i = 0; i<queries.length;i++){
            int count = 0;
            int sum = 0;
            while (count<nums.length && sum+nums[count] <= queries[i]){
                sum+=nums[count];
                count++;
            }
            results[i] = count;
        }
        return results;
    }
}

File: 2389.longestSubsequenceWithLimitedSum.ts
/*
Runtime 96 ms Beats 88.24%
Memory 46.1 MB Beats 47.6%
*/

function answerQueries(nums: number[], queries: number[]): number[] {
    nums.sort((a,b)=>{return a-b})
    const result:number[] = []
    for (let num of queries){
        let count = 0
        let sum = 0
        while (sum+nums[count] <= num && count < nums.length){
            sum+=nums[count]
            count++
        }
        result.push(count)
    }
    return result
};